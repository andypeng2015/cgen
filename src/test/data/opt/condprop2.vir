module foo

export function w $foo(w %x) {
@start:
  %c = lt.w 1_w, %x
  br %c, @y, @n
@y:
  %r = copy.w 1_w
  jmp @end
@n:
  %r = flag.w %c
  jmp @end
@end:
  ret %r
}
