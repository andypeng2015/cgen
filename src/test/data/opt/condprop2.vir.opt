module foo

export function w $foo(w %x) {
@2:
  %3 = gt.w %x, 0x1_w ; @12
  %4 = flag.w %3 ; @13
  ret %4
}
