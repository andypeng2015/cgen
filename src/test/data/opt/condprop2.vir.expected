module foo

export function w $foo(w %x) {
@5:
  %0 = gt.w %x, 0x1_w ; @7
  br %0, @0(0x1_w), @0(0x0_w)
@0(%r.3):
  ret %r.3
}
